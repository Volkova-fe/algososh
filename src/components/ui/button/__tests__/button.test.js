import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';

import { Button } from '../button';

describe('Тестирование компонента Button', () => {

	it('Кнопка c текстом рендерится без ошибок', () => {
		const button = TestRenderer
			.create(<Button text='Текст' />)
			.toJSON()
		expect(button).toMatchSnapshot()
	})

	it('Кнопка без текста рендерится без ошибок', () => {
		const button = TestRenderer
			.create(<Button />)
			.toJSON()
		expect(button).toMatchSnapshot()
	})

	it('Кнопка заблокирована, рендерится без ошибок', () => {
		const button = TestRenderer
			.create(<Button disabled />)
			.toJSON()
		expect(button).toMatchSnapshot()
	})

	it('Кнопка c индикацией загрузки рендерится без ошибок', () => {
		const button = TestRenderer
			.create(<Button isLoader={true} />)
			.toJSON()
		expect(button).toMatchSnapshot()
	})

	it('Нажатие на кнопку вызывает колбек', () => {
		window.alert = jest.fn();

		render(<Button text='Вызов колбека' onClick={() => { alert('Успешный вызов колбека') }} />)

		const button = screen.getByText("Вызов колбека");
		fireEvent.click(button);

		expect(window.alert).toHaveBeenCalledWith('Успешный вызов колбека');
	});
})
