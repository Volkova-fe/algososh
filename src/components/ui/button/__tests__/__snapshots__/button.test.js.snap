// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента Button Кнопка c индикацией загрузки рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  data-testid="button"
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Тестирование компонента Button Кнопка c текстом рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  data-testid="button"
  type="button"
>
  <p
    className="text undefined"
  >
    Текст
  </p>
</button>
`;

exports[`Тестирование компонента Button Кнопка без текста рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  data-testid="button"
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование компонента Button Кнопка заблокирована, рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  data-testid="button"
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
